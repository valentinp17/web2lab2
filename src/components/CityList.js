import React from 'react';
import LocationWeatherInfo from "./LocationWeatherInfo";
import DeleteCity from "./DeleteCity";
import AddCity from "./AddCity";
import {connect} from "react-redux";
import {addCity} from "../actions/AddCity";
import {deleteCity} from "../actions/DeleteCity";
import {updateWeather} from "../actions/FetchCity";
import '../styles/CityList.css'

function mapDispatchToProps(dispatch) {
    return {
        addCity: city => dispatch(addCity(city)),
        deleteCity: city => dispatch(deleteCity(city)),
        updateWeather: city => dispatch(updateWeather(city))
    };
}

function mapStateToProps(state) {
    return {cities: state.cities};
};

class ConnectedCityList extends React.Component {


    addCity = (cityName) => {
        const timeAdded = Date.now();
        this.props.addCity({
            name: cityName,
            timeAdded: timeAdded
        });
    }

    updateWeather = (city) => {
        this.props.updateWeather(city);
    }

    removeCity = (city) => {
        this.props.deleteCity(city);
    }

    formatCities = (cities) => {
        return cities.map((city) => {
            return <li key={city.timeAdded}>
                    <LocationWeatherInfo city={city}/>
                    <DeleteCity city={city} removeCity={this.removeCity}/>
                </li>
            }
        );
    };

    componentDidMount() {
        this.props.cities.map((city) => {
            this.updateWeather({city: city});
            return null;
        })
    }


    render() {
        return (
            <div>
                <AddCity addCity={this.addCity}/>
                <ul id="city-grid">{this.formatCities(this.props.cities)}</ul>
            </div>

        );
    }
}

const CityList = connect(
    mapStateToProps,
    mapDispatchToProps
)(ConnectedCityList);

export default CityList
