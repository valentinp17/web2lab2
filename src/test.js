import renderer from "react-test-renderer";
import React from "react";
import {configure, shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from "redux-mock-store";

import DeleteCity from './components/DeleteCity'
import AddCity from "./components/AddCity";
import DefaultCityInfo from "./components/DefaultCityInfo";
import CityList from "./components/CityList";
import LocationWeatherInfo from "./components/LocationWeatherInfo";

configure({adapter: new Adapter()});

test("should render DeleteCity", () => {
    const tree = shallow(<DeleteCity/>);
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render AddCity", () => {
    const tree = shallow(<AddCity/>);
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render DefaultCityInfo without geo", () => {
    const mockStore = configureMockStore();
    const store = mockStore({
        defaultCity: {
            name: 'санкт-петербург',
            latitude: undefined,
            longitude: undefined
        }
    });

    const tree = shallow(<DefaultCityInfo store={store}/>).dive();
    tree.render();
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render DefaultCityInfo with geo", () => {
    const mockStore = configureMockStore();
    const store = mockStore({
        defaultCity: {
            name: 'санкт-петербург',
            latitude: 55,
            longitude: 55
        }
    });

    const tree = shallow(<DefaultCityInfo store={store}/>).dive();
    tree.render();
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render CityList with 2 cities", () => {
    const mockStore = configureMockStore();
    const store = mockStore({
        "cities": [{
            "temp": -1.1,
            "name": "Moscow",
            "timeAdded": 1577384881715,
            "pressure": 1009,
            "humidity": 100,
            "wind": 2,
            "icon": "50n",
            "isLoading": false
        }, {
            "temp": 25.66,
            "name": "Kek",
            "timeAdded": 1577384888217,
            "pressure": 1007,
            "humidity": 29,
            "wind": 0.3,
            "icon": "03n",
            "isLoading": false
        }]
    });

    const tree = shallow(<CityList store={store}/>).dive();
    tree.render();
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render CityList with no cities", () => {
    const mockStore = configureMockStore();
    const store = mockStore({
        "cities": []
    });

    const tree = shallow(<CityList store={store}/>).dive();
    tree.render();
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render LocationWeatherInfo with loading", () => {
    const tree = shallow(<LocationWeatherInfo city={{
        "timeAdded": 1577384888217,
        "isLoading": true
    }}/>);
    expect(toJson(tree)).toMatchSnapshot();
});
test("should render LocationWeatherInfo with weather", () => {
    const tree = shallow(<LocationWeatherInfo city={{
        "temp": -1.1,
        "name": "Moscow",
        "timeAdded": 1577384881715,
        "pressure": 1009,
        "humidity": 100,
        "wind": 2,
        "icon": "50n",
        "isLoading": false
    }}/>);
    expect(toJson(tree)).toMatchSnapshot();
});

test("should render LocationWeatherInfo with weather for main city", () => {
    const tree = shallow(<LocationWeatherInfo city={{
        "temp": 0.58,
        "name": "Kolomyagi",
        "pressure": 1013,
        "humidity": 86,
        "wind": 3,
        "icon": "04n",
        "isLoading": false,
        "longitude": 30.2557549,
        "latitude": 60.02116389999999
    }}/>);
    expect(toJson(tree)).toMatchSnapshot();
});


