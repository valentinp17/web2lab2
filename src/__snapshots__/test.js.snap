// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddCity 1`] = `
<div
  className="addCity"
>
  <form
    onSubmit={[Function]}
  >
    <input
      placeholder="Type the city here"
    />
    <button>
      add city
    </button>
  </form>
</div>
`;

exports[`should render CityList with 2 cities 1`] = `
<ConnectedCityList
  addCity={[Function]}
  cities={
    Array [
      Object {
        "humidity": 100,
        "icon": "50n",
        "isLoading": false,
        "name": "Moscow",
        "pressure": 1009,
        "temp": -1.1,
        "timeAdded": 1577384881715,
        "wind": 2,
      },
      Object {
        "humidity": 29,
        "icon": "03n",
        "isLoading": false,
        "name": "Kek",
        "pressure": 1007,
        "temp": 25.66,
        "timeAdded": 1577384888217,
        "wind": 0.3,
      },
    ]
  }
  deleteCity={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateWeather={[Function]}
/>
`;

exports[`should render CityList with no cities 1`] = `
<ConnectedCityList
  addCity={[Function]}
  cities={Array []}
  deleteCity={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateWeather={[Function]}
/>
`;

exports[`should render DefaultCityInfo with geo 1`] = `
<ConnectedDefaultCityInfo
  defaultCity={
    Object {
      "latitude": 55,
      "longitude": 55,
      "name": "санкт-петербург",
    }
  }
  getLocation={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`should render DefaultCityInfo without geo 1`] = `
<ConnectedDefaultCityInfo
  defaultCity={
    Object {
      "latitude": undefined,
      "longitude": undefined,
      "name": "санкт-петербург",
    }
  }
  getLocation={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`should render DeleteCity 1`] = `
<div
  className="delete"
>
  <button
    className="deleteButton"
    onClick={[Function]}
  >
    delete city
  </button>
</div>
`;

exports[`should render LocationWeatherInfo with loading 1`] = `
<div
  key="true"
>
  <div />
  <div>
    <p>
      Грузится...
    </p>
  </div>
</div>
`;

exports[`should render LocationWeatherInfo with weather 1`] = `
<div
  key="false"
>
  <div />
  <div
    className="weather"
  >
    <div
      className="city"
    >
      Moscow
    </div>
    <div
      className="img"
    >
      <img
        alt="weather icon"
        src="http://openweathermap.org/img/wn/50n@2x.png"
      />
    </div>
    <div
      className="infoType"
    >
      <div>
        temperature
      </div>
      <div>
        -1.1
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        humidity
      </div>
      <div>
        100
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        pressure
      </div>
      <div>
        1009
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        wind
      </div>
      <div>
        2
      </div>
    </div>
  </div>
</div>
`;

exports[`should render LocationWeatherInfo with weather for main city 1`] = `
<div
  key="false"
>
  <div />
  <div
    className="weather"
  >
    <div
      className="defaultCity"
    >
      Kolomyagi
    </div>
    <div
      className="img"
    >
      <img
        alt="weather icon"
        src="http://openweathermap.org/img/wn/04n@2x.png"
      />
    </div>
    <div
      className="infoType"
    >
      <div>
        temperature
      </div>
      <div>
        0.58
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        humidity
      </div>
      <div>
        86
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        pressure
      </div>
      <div>
        1013
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        wind
      </div>
      <div>
        3
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        longitude
      </div>
      <div>
        30.26
      </div>
    </div>
    <div
      className="infoType"
    >
      <div>
        latitude
      </div>
      <div>
        60.02
      </div>
    </div>
  </div>
</div>
`;
